File: Benchmarking/src/gen22.cpp
  #include <vector>
  #include <iostream>
  #include <chrono>
  struct s0 {
    std::vector<std::vector<double>> p0;
    std::vector<std::vector<double>> p1;
    s0(std::vector<std::vector<double>> v0, std::vector<std::vector<double>> v1): p0(v0), p1(v1) { }
  };
  struct s1 {
    std::vector<s0> p0;
    s0 p1;
    s1(std::vector<s0> v0, s0 v1): p0(v0), p1(v1) { }
  };
  double f23(const double &v0) {
    double v7 = v0;
    double v5 = v0;
    double v2 = v5;
    double v1 = v2;
    double v6 = v7;
    const double v4 = v6 / v7;
    double v8 = v4;
    double v19 = v4;
    v6 = v1;
    const double v15 = v8 / v19;
    const std::vector<double> v9 { v1, v6, v15, v19, v8, v1, v0 };
    std::vector<double> v35 = v9;
    const double v13 = v35[1];
    const std::vector<double> v25 { v1 };
    const double v31 = v25[0];
    const std::vector<std::vector<double>> v24 { v35, v35 };
    v35[6] = v13;
    const std::vector<double> v49 = v24[1];
    std::vector<double> v12 = v25;
    v35[5] = v4;
    const double v75 = v12[0];
    const double v43 = v49[1];
    v7 = v31;
    v35[3] = v75;
    return v43;
  }
  __attribute__((noinline))
  double f0(const std::vector<s1> &v0, const s0 &v1, const double &v2) {
    double v36 = v2;
    double v22 = v36;
    const double v15 = f23(v2);
    const double v40 = f23(v15);
    const double v25 = f23(v40);
    const double v28 = v2 / v22;
    const double v42 = v28 + v25;
    v36 = v28;
    return v42;
  }
  int main() {
    std::vector<s1> v0({ { { { { { 0.0 } }, { { 1.0 } } } }, { { { 2.0 } }, { { 3.0 } } } } });
    s0 v1({ { { 4.0 } }, { { 5.0 } } });
    double v2(6.0);
    auto start = std::chrono::high_resolution_clock::now();
    double result;
    for (int i = 0; i < 1000; i ++) {
      result = f0(v0, v1, v2);
    }
    auto end = std::chrono::high_resolution_clock::now();
    double value = *((double*) &result);
    std::cout << value << "\n";
    std::cout << std::chrono::duration_cast<std::chrono::nanoseconds>(end-start).count();
    std::cout << "\n";
    return 0;
  }

File: Benchmarking/src/gen22.swift
  import Dispatch
  struct s0 {
    var p0: [[Double]]
    var p1: [[Double]]
  }
  struct s1 {
    var p0: [s0]
    var p1: s0
  }
  func f23(_ v0: Double) -> Double {
    var v7: Double = v0
    var v5: Double = v0
    var v2: Double = v5
    var v1: Double = v2
    var v6: Double = v7
    let v4: Double = v6 / v7
    var v8: Double = v4
    var v19: Double = v4
    v6 = v1
    let v15: Double = v8 / v19
    let v9: [Double] = [v1, v6, v15, v19, v8, v1, v0]
    var v35: [Double] = v9
    let v13: Double = v35[1]
    let v25: [Double] = [v1]
    let v31: Double = v25[0]
    let v24: [[Double]] = [v35, v35]
    v35[6] = v13
    let v49: [Double] = v24[1]
    var v12: [Double] = v25
    v35[5] = v4
    let v75: Double = v12[0]
    let v43: Double = v49[1]
    v7 = v31
    v35[3] = v75
    return v43
  }
  @inline(never)
  func f0(_ v0: [s1], _ v1: s0, _ v2: Double) -> Double {
    var v36: Double = v2
    var v22: Double = v36
    let v15: Double = f23(v2)
    let v40: Double = f23(v15)
    let v25: Double = f23(v40)
    let v28: Double = v2 / v22
    let v42: Double = v28 + v25
    v36 = v28
    return v42
  }
  func benchmark() {
    let v0: [s1] = [s1(p0: [s0(p0: [[0.0]], p1: [[1.0]])], p1: s0(p0: [[2.0]], p1: [[3.0]]))]
    let v1: s0 = s0(p0: [[4.0]], p1: [[5.0]])
    let v2: Double = 6.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 6.0
    for _ in 1...1000 {
      result = f0(v0, v1, v2)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()
